### PR TITLE
Update deprecated LibExpat method ParsedData to ETree and minimum version # for dep warns in other packages

### PR DESCRIPTION
**Pull request by musm**
Needs a bump in LibExpat require after the macro hygiene bugfix

### CODE REVIEW

File: src/WinRPM.jl [hunk 1]
```diff
@@ -55,7 +55,7 @@ if is_unix()
     end
 elseif is_windows()
     function download(source::AbstractString; retry = 5)
-        dest = Array(UInt16,261)
+        dest = Array{UInt16}(261)
```
Comment (tkelman) [inline src/WinRPM.jl]: include the dimensionality `Array{UInt16,1}` or `Vector{UInt16}`

**Comment by musm** [comment]
fxd

**Comment by musm** [comment]
I've bumped LibExpat to 0.2.3  so please  do not merge untill  https://github.com/JuliaLang/METADATA.jl/pull/7565 is tagged

this will help get rid off  more 0.6 dep warns

**Comment by musm** [comment]
g2g now?

**Comment by tkelman** [comment]
the minimum versions are too tight for 0.4 and 0.5, they rule out working versions there

**Comment by musm** [comment]
but depwarns hurt user experience and are slow

**Comment by tkelman** [comment]
and on 0.6 it's a better experience to use the latest versions of things. but they're not required to work on the rest of the julia versions supported here

**Comment by musm** [comment]
but there won't be depwarns on 0.4, and 0.5 with this PR

**Comment by tkelman** [comment]
the code change is fine, just the minimum dependencies are too high

**Comment by musm** [comment]
Libz and URIParser won't work on 0.6 without those version because of the `.data` string  changes.  LibExpat won't work without 0.2.3

**Comment by tkelman** [comment]
and when this package is 0.6-only, those can be the minimum versions

**Comment by tkelman** [comment]
we can upper bound all past versions of those packages to 0.6 if they're all broken there. when was the ETree name introduced in LibExpat.jl?

**Comment by musm** [comment]
So upper bound julia  to `julia 0.4 0.6-` in this PR?

**Comment by musm** [comment]
tag a version. Then on master update the require to `julia 0.6-` and update all the minimums?

**Comment by tkelman** [comment]
I meant upper bound the old versions of the other packages, where they are known to be broken. It would be a bit of a hassle to need to maintain separate branches for this package just yet, not actually worth making it 0.6-only right now.

File: src/WinRPM.jl [hunk 2]
```diff
@@ -50,8 +50,8 @@ if is_unix()
         unsafe_string(x.body), x.http_code
     end
 elseif is_windows()
-    function download(source::AbstractString; retry=5)
-        dest = Array(UInt16,261)
+  function download(source::AbstractString; retry=5)
```
Comment (tkelman) [inline src/WinRPM.jl]: don't change indent here

**Comment by musm** [comment]
So I go to Libz and LibExpat and URIParser and upper bound the version right before it works on 0.6  to be `julia 0.4(or lower)  0.6-`     What about the REQUIRE in this package

File: src/WinRPM.jl [hunk 3]
```diff
@@ -199,10 +199,9 @@ Package(p::Vector{ETree}) = [Package(pkg) for pkg in p]
 
 getindex(pkg::Package,x) = getindex(pkg.pd, x)
 
-@compat immutable Packages{T<:Union{Set{ETree},Vector{ETree},}}
+immutable Packages{T<:Union{Set{ETree},Vector{ETree}}}
     p::T
 end
-@compat Packages{T<:Union{Set{ETree},Vector{ETree}}}(pkgs::T) = Packages{T}(pkgs)
```
Comment (musm) [inline src/WinRPM.jl]: redundant with l202

File: src/WinRPM.jl [hunk 4]
```diff
@@ -199,10 +199,9 @@ Package(p::Vector{ETree}) = [Package(pkg) for pkg in p]
 
 getindex(pkg::Package,x) = getindex(pkg.pd, x)
 
-@compat immutable Packages{T<:Union{Set{ETree},Vector{ETree},}}
```
Comment (musm) [inline src/WinRPM.jl]: weird comma after ETree

**Comment by musm** [comment]
I won't make any of the REQUIRE changes.  I just want to confirm that Pkg.update will grab the latest version of the package and all of its dependencies that can be satisfied. It that case it should be ok, even though suboptimal.

**Pull request merged**